rfcs/2020-08-26-3191-host-metrics.md

```diff
@@ -0,0 +1,204 @@
+- `filesystem_size_bytes` tagged with the device, filesystem type, and mountpoint (gauge)
+- `filesystem_total_file_nodes` tagged with the device, filesystem type, and mountpoint (gauge)
+- `filesystem_free_file_nodes` tagged with the device, filesystem type, and mountpoint (gauge)
+- `load1` (gauge)
+- `load5` (gauge)
+- `load15` (gauge)
+- `memory_active_bytes` (gauge)
+- `memory_compressed_bytes` (gauge)
+- `memory_free_bytes` (gauge)
+- `memory_inactive_bytes` (gauge)
+- `memory_swap_total_bytes` (gauge)
+- `memory_swap_used_bytes` (gauge)
+- `memory_swapped_in_bytes_total` (gauge)
+- `memory_swapped_out_bytes_total` (gauge)
+- `memory_total_bytes` (gauge)
+- `memory_wired_bytes` (gauge)
+- `network_receive_bytes_total` tagged with device (counter)
+- `network_receive_errs_total` tagged with device (counter)
+- `network_receive_multicast_total` tagged with device (counter)
+- `network_receive_packets_total` tagged with device (counter)
+- `network_transmit_bytes_total` tagged with device (counter)
+- `network_transmit_errs_total` tagged with device (counter)
+- `network_transmit_multicast_total` tagged with device (counter)
+- `network_transmit_packets_total` tagged with device (counter)
+
+Users should be able to limit the collection of metrics to specific classes, here: `cpu`, `memory`, `disk`, `filesystem`, `load`, and `network`.
+
+Metrics will also be tagged with:
+
+- `host`: the host name of the host being monitored.
+
+And `collector` for type of metric:
+
+- `disk` for disk based metrics
+- `cpu` for CPU based metrics
+- `filesystem` for filesystem based metrics
+- `memory` for memory based metrics.
+- `load` for load based metrics.
+- `network` for network based metrics.
+
+Specific explanation of some of the filesystem metrics:
+
+- `filesystem_avail_bytes` = Filesystem space available to non-root users in bytes (including reserved blocks).
+- `filesystem_free_bytes` = Filesystem free space in bytes (excluding reserved blocks).
+- `filesystem_size_bytes` = Filesystem size in bytes.
+
+
+## Doc-level Proposal
+
+The following additional source configuration will be added:
+
+```toml
+[sources.my_source_id]
+  type = "host_metrics" # required
+  collectors = [ "all"] # optional, defaults collecting all metrics.
+  filesystem.mountpoints = [ "*" ] # optional, defaults to all mountpoints.
+  disk.devices = [ "*" ] # optional, defaults to all to disk devices.
+  network.devices = [ "*" ] # optional, defaults to all to network devices.
+  scrape_interval_secs = 15 # optional, default, seconds
+  namespace = "host" # optional, default is "host", namespace to put metrics under
+```
+
+For `collector` (name open to discussion) we'd specify an array of metric collectors:
+
+```toml
+collectors = [ "cpu", "memory", "network" ]
+```
+
+For disk and network devices or filesystem mountpoints the default is to collect for all ("*") devices and mountpoints. Or you can configure Vector to only collect from specific devices, for example:
```

Comment:
I'm wondering if will be useful to also allow users to specify a denylist rather than allowlist, in the future. Maybe we could model this like `network.devices = [ "!eth0", "*"]` to monitor everything but eth0? We could expand this pattern generally to this type of "filter" config.